keyboards/keyboardio/model01/keymaps/pugmajere/keymap.c

```diff
@@ -0,0 +1,186 @@
+  KC_PGDN, KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   , KC_ESC ,    KC_RALT, KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_MINS,
+                                  LCTL_T(KC_LBRC),                                 RCTL_T(KC_RBRC),
+                                            KC_BSPC,                                     KC_SPC ,
+                                          KC_MT_LCBRC,                              KC_MT_RCBRC,
+                                             KC_MT_LPAR,                           KC_MT_RPAR,
+                                                   MO(FUN),                         MO(FUN)
+  ),
+[FUN] = LAYOUT(
+  _______, KC_F1  , KC_F2  , KC_F3  , KC_F4  , KC_F5  ,                      KC_F6  , KC_F7  , KC_F8  , KC_F9  , KC_F10 , KC_F11 ,
+  KC_TAB , _______, KC_MS_U, KC_WH_U, KC_BTN3, MACRO_KVM_1, RGB_TOG,    KC_MPRV, KC_MNXT, KC_LCBR, KC_RCBR, KC_LBRC, KC_RBRC, KC_F12 ,
+  KC_HOME, KC_MS_L, KC_MS_D, KC_MS_R, KC_BTN1, MACRO_KVM_2, RGB_SPI,    KC_MPLY, KC_LEFT, KC_DOWN, KC_UP  , KC_RGHT, _______, _______,
+  KC_END , KC_PSCR, KC_INS , KC_WH_D, KC_BTN2, MACRO_KVM_4, RGB_SPD,    _______, KC_MUTE, KC_VOLD, KC_VOLU, _______, KC_BSLS, KC_PIPE,
+                                         _______,                                 _______,
+                                            KC_DEL ,                           KC_ENT ,
+                                               _______,                     _______,
+                                                  _______,               _______,
+                                      _______,                                        _______
+  )
+};
+
+/* template for new layouts:
+LAYOUT(
+  _______, _______, _______, _______, _______, _______,                      _______, _______, _______, _______, _______, _______,
+  _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______,
+  _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______,
+  _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______,
+                                         _______,                                 _______,
+                                            _______,                           _______,
+                                               _______,                     _______,
+                                                  _______,               _______,
+                                      _______,                                        _______
+  )
+*/
+// clang-format on
+
+void mod_type(uint16_t modcode, uint16_t keycode) {
+    register_mods(MOD_BIT(modcode));
+    register_code(keycode);
+    unregister_code(keycode);
+    unregister_mods(MOD_BIT(modcode));
+}
```

Comment:
`tap_code16` does this, explicitly.

Reply:
Oh! Thanks, that is a bit of code I didn't really want to have in here, I just didn't find tap_code16 when I was puzzling this out.